src/ddsrt/src/xmlparser.c

```diff
@@ -377,22 +377,27 @@ static void discard_payload (struct ddsrt_xmlp_state *st)
   st->tpescp = 0;
 }
 
-static int append_to_payload (struct ddsrt_xmlp_state *st, int c, int islit)
+static void append_literal_to_payload (struct ddsrt_xmlp_state *st, int c)
 {
-  if (!islit) {
-    st->tp[st->tpp++] = (char) c;
-  } else {
-    if (st->tpescp < st->tpp) {
-      size_t n = st->tpp - st->tpescp;
-      if (unescape_insitu (st->tp + st->tpescp, &n) < 0) {
-        discard_payload (st);
-        return -1;
-      }
-      st->tpp = st->tpescp + n;
+  st->tp[st->tpp++] = (char) c;
+  if (st->tpp == st->tpsz) {
+    st->tpsz += 1024;
+    st->tp = ddsrt_realloc (st->tp, st->tpsz);
+  }
+}
+
+static int append_to_payload (struct ddsrt_xmlp_state *st, int c)
+{
+  if (st->tpescp < st->tpp) {
+    size_t n = st->tpp - st->tpescp;
+    if (unescape_insitu (st->tp + st->tpescp, &n) < 0) {
+      discard_payload (st);
+      return -1;
     }
-    st->tp[st->tpp++] = (char) c;
-    st->tpescp = st->tpp;
+    st->tpp = st->tpescp + n;
   }
+  st->tp[st->tpp++] = (char) c;
```

Comment:
For this I would rely on (the proposed new) `append_literal_to_payload`. The only thing is to remember to retain the update of `st->tpescp` 🙂

Reply:
Yup, I've swapped to using `append_literal_to_payload` in `append_to_payload`. I thought about doing this before but it slipped my mind (or perhaps I was just lazy :stuck_out_tongue_closed_eyes:). Either way thanks for catching this.